app/src/features/reports/CompareStatesForVariableReport.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import WithDatasets from "../../utils/WithDatasets";
+import { LinkWithStickyParams } from "../../utils/urlutils";
+import VerticalGroupedBarChart from "../charts/VerticalGroupedBarChart";
+import StackedBarChart from "../charts/StackedBarChart";
+import { Button, Grid } from "@material-ui/core";
+import useDatasetStore from "../../utils/useDatasetStore";
+import variableProviders, {
+  VariableProvider,
+} from "../../utils/variableProviders";
+import { Breakdowns } from "../../utils/DatasetTypes";
+
+function getUniqueDatasetIds(providers: VariableProvider[]): string[] {
+  return Array.from(new Set(providers.map((p) => p.datasetIds).flat()));
+}
+
+const ByStateAndRace: Breakdowns = {
+  geography: "state",
+  demographic: "race",
+};
+
+const ByRace: Breakdowns = {
+  geography: "national",
+  demographic: "race",
+};
+
```

Comment:
It seems like these (getUniqueDatasetIds, ByStateAndRace, ByRace) could be reused elsewhere as well? is it worth putting them somewhere common?

Reply:
Good call! I was going to do this later but I went ahead and did it now. I made Breakdowns into a class and added some factory/builder methods to allow constructing it dynamically in a more readable way, should be pretty reusable now. And I added getUniqueDatasetIds to the base class.